src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -1309,7 +1309,12 @@ public function registerForAutoconfiguration(string $interface)
     /**
      * Registers an attribute that will be used for autoconfiguring annotated classes.
      *
-     * The configurator will receive a ChildDefinition instance, an instance of the attribute and the corresponding \ReflectionClass, in that order.
+     * The configurator will receive (in this order):
+     *  - a ChildDefinition instance
+     *  - the attribute instance
+     *  - the corresponding reflector (\ReflectionClass|\ReflectionMethod|\ReflectionProperty|\ReflectionParameter)
+     *
+     * If an attribute can be used on multiple targets, a union type should be used.
```

Comment:
```suggestion
     * If an attribute can be used on multiple targets, a union type should be used.
     *
     * @param callable(ChildDefinition, object $attribute, \ReflectionClass|\ReflectionMethod|\ReflectionProperty|\ReflectionParameter):void $configurator
```

@Tobion Like this?

Reply:
Yes. Then there is no need for the proposa "The configurator will receive (in this order): ...".
Would be even better to tell static analysis that the `$attribute` param is an instance of `$attributeClass`. Should also be possible.

Reply:
also `@param class-string $attributeClass` would be good

Reply:
Applied your feedback.